epoch timestamp is rendered to wrong date

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Is there an issue with rendering custom date format from epoch?

Example config:

```
❯ cat ~/.config/omp/omp-date.toml
version = 2
final_space = true

[[blocks]]
type = "prompt"
alignment = "left"
newline = false

[[blocks.segments]]
type = "text"
style = "plain"
template = '{{ $n1 := (now|unixEpoch) }}{{ $n2 := (now|date_modify "+168h"|unixEpoch) }}{{ $n1 }} {{ $n1 |date "Mon 2/Jan 15:04" }} || {{ $n2 }} {{ $n2 |date "Mon 2/Jan 15:04" }}'
``` 

Unexpected result: see the epoch values show the 1w diff, but the formatted dates both have same day:

```
❯ oh-my-posh print preview --config ~/.config/omp/omp-date.toml

Primary:

1776455769 Fri 17/Apr 21:56 || 1777060569 Fri 17/Apr 21:56
```

OMP version:

```
❯ oh-my-posh --version
29.10.0
```


### Theme

none

### What OS are you seeing the problem on?

macOS

### Which shell are you using?

zsh

### Log output

```Shell
oh-my-posh debug --plain --config ~/.config/omp/omp-date.toml

Version: 29.10.0

Shell: shell (5.9)

Prompt:

1776456341 Fri 17/Apr 22:05 || 1777061141 Fri 17/Apr 22:05

Segments:

ConsoleTitle(false)                       -   0 ms
Text(true)                                -   0 ms

Run duration: 11.158834ms

Cache path: /Users/jkovacs/.cache/oh-my-posh

Config path: /Users/jkovacs/.config/omp/omp-date.toml

Logs:

[DEBUG] 22:05:41.676 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 22:05:41.676 store.go:]:163 → (session) key not found: prompt_count_cache
[TRACE] 22:05:41.676 store.go:](session prompt_count_cache) - 15.792µs
[TRACE] 22:05:41.676 terminal.go:setPromptCount() - 18.833µs
[TRACE] 22:05:41.676 terminal.go:GOOS() - 41ns
[DEBUG] 22:05:41.676 terminal.go:setPwd:102 → /Users/jkovacs
[TRACE] 22:05:41.676 terminal.go:setPwd() - 11.125µs
[TRACE] 22:05:41.676 terminal.go:Init() - 39.125µs
[DEBUG] 22:05:41.676 home.go:12 → /Users/jkovacs
[TRACE] 22:05:41.676 path.go:Path() - 21.875µs
[DEBUG] 22:05:41.676 store.go:init:95 → (device) loading DSC_CONFIGURATION
[DEBUG] 22:05:41.676 store.go:init:95 → (device) loading environment_platform
[DEBUG] 22:05:41.676 store.go:init:95 → (device) loading DSC_SHELL
[DEBUG] 22:05:41.676 store.go:init:95 → (device) loading is_wsl
[DEBUG] 22:05:41.676 store.go:init:95 → (device) loading INITVERSIONZSH
[TRACE] 22:05:41.676 store.go:init(device omp.cache) - 225.875µs
[TRACE] 22:05:41.676 init.go:SessionID() - 500ns
[TRACE] 22:05:41.676 path.go:Path() - 42ns
[DEBUG] 22:05:41.676 store.go:init:95 → (session) loading CONFIG
[DEBUG] 22:05:41.676 store.go:init:95 → (session) loading prompt_count_cache
[DEBUG] 22:05:41.676 store.go:init:95 → (session) loading template_cache
[DEBUG] 22:05:41.676 store.go:init:95 → (session) loading toggle_cache
[DEBUG] 22:05:41.676 store.go:init:95 → (session) loading CONFIG_SOURCE
[TRACE] 22:05:41.677 store.go:init(session zsh.25533202-522b-4ca8-ae70-fb78f4ceb2d5.omp.cache) - 97.5µs
[DEBUG] 22:05:41.677 load.go:isTheme:384 → /Users/jkovacs/.config/omp/omp-date.toml is not a theme
[TRACE] 22:05:41.677 load.go:resolveConfigLocation() - 13.125µs
[DEBUG] 22:05:41.677 store.go:]:176 → (device) found entry: DSC_CONFIGURATION - [0x449c191de900 0x449c191dea80 0x449c191dec00 0x449c191ded80]
[TRACE] 22:05:41.677 store.go:](device DSC_CONFIGURATION) - 23.917µs
[DEBUG] 22:05:41.677 home.go:12 → /Users/jkovacs
[DEBUG] 22:05:41.677 resource.go:Add:47 → adding item
[TRACE] 22:05:41.677 load.go:read() - 115.667µs
[DEBUG] 22:05:41.677 store.go:]:176 → (session) found entry: toggle_cache - map[]
[TRACE] 22:05:41.677 store.go:](session toggle_cache) - 8.541µs
[DEBUG] 22:05:41.677 store.go:]:199 → (session) setting entry: toggle_cache - map[] with duration: infinite
[TRACE] 22:05:41.677 store.go:](session toggle_cache) - 4.875µs
[DEBUG] 22:05:41.677 store.go:]:199 → (device) setting entry: DSC_CONFIGURATION - [0x449c191de900 0x449c191dea80 0x449c191dec00 0x449c191ded80 0x449c191def00] with duration: infinite
[TRACE] 22:05:41.677 store.go:](device DSC_CONFIGURATION) - 4.375µs
[TRACE] 22:05:41.677 load.go:Parse() - 200.333µs
[TRACE] 22:05:41.677 load.go:Load() - 202.167µs
[TRACE] 22:05:41.677 terminal.go:Shell() - 84ns
[TRACE] 22:05:41.677 terminal.go:Flags() - 42ns
[DEBUG] 22:05:41.677 store.go:]:176 → (session) found entry: template_cache - {map[Path:0x449c191da480 Session:0x449c1937e120 Text:0x449c1937e030 Time:0x449c1930cb90] map[] ~ jkovacs  jkovacs macbook 5.9 zsh /Users/jkovacs darwin 0.0.0-dev 434 1 0 0 false false}
[TRACE] 22:05:41.677 store.go:](session template_cache) - 12.708µs
[TRACE] 22:05:41.677 cache.go:restoreCache() - 15.083µs
[DEBUG] 22:05:41.677 writer.go:Init:107 → terminal program: ghostty
[DEBUG] 22:05:41.677 writer.go:Init:108 → terminal shell: shell
[DEBUG] 22:05:41.677 terminal.go:Getenv:64 → NO DATA
[TRACE] 22:05:41.677 terminal.go:Getenv(OMP_CACHE_DISABLED) - 1.708µs
[DEBUG] 22:05:41.677 store.go:]:163 → (device) key not found: accent_color
[TRACE] 22:05:41.677 store.go:](device accent_color) - 4.542µs
[ERROR] 22:05:41.687 terminal.go:RunCommand:277 → exit status 1
[DEBUG] 22:05:41.687 terminal.go:RunCommand:280 ↓
    2026-04-17 22:05:41.686 defaults[59618:12547268]
    The domain/default pair of (kCFPreferencesAnyApplication, AppleAccentColor) does not exist
[TRACE] 22:05:41.687 terminal.go:RunCommand(defaults read -g AppleAccentColor) - 10.03375ms
[ERROR] 22:05:41.687 colors_darwin.go:GetAccentColor:14 → exit status 1
[TRACE] 22:05:41.687 colors.go:SetAccentColor() - 10.051709ms
[TRACE] 22:05:41.687 terminal.go:Shell() - 84ns
[DEBUG] 22:05:41.687 terminal.go:Getenv:64 → 5.9
[TRACE] 22:05:41.687 terminal.go:Getenv(POSH_SHELL_VERSION) - 2.875µs
[DEBUG] 22:05:41.687 debug.go:PrintDebug:24 → segment: Title
[TRACE] 22:05:41.687 terminal.go:Flags() - 41ns
[DEBUG] 22:05:41.687 terminal.go:Getenv:64 → NO DATA
[TRACE] 22:05:41.687 terminal.go:Getenv(POSH_CURSOR_LINE) - 1.708µs
[DEBUG] 22:05:41.687 terminal.go:Getenv:64 → NO DATA
[TRACE] 22:05:41.687 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 1.125µs
[TRACE] 22:05:41.687 terminal.go:Flags() - 0s
[TRACE] 22:05:41.687 terminal.go:Flags() - 0s
[TRACE] 22:05:41.687 terminal.go:Flags() - 0s
[TRACE] 22:05:41.687 terminal.go:Flags() - 209ns
[DEBUG] 22:05:41.687 home.go:12 → /Users/jkovacs
[TRACE] 22:05:41.687 terminal.go:GOOS() - 83ns
[DEBUG] 22:05:41.687 segment.go:Execute:173 → segment: Text
[DEBUG] 22:05:41.687 store.go:]:176 → (session) found entry: toggle_cache - map[]
[TRACE] 22:05:41.687 store.go:](session toggle_cache) - 5.583µs
[DEBUG] 22:05:41.687 segment.go:isToggled:291 → no toggles found
[TRACE] 22:05:41.687 terminal.go:Flags() - 0s
[DEBUG] 22:05:41.687 text.go:patchTemplate:149 → {{ $n1 := (now|unixEpoch) }}{{ $n2 := (now|date_modify "+168h"|unixEpoch) }}{{ $n1 }} {{ $n1 |date "Mon 2/Jan 15:04" }} || {{ $n2 }} {{ $n2 |date "Mon 2/Jan 15:04" }}
[TRACE] 22:05:41.687 terminal.go:Shell() - 42ns
[TRACE] 22:05:41.687 terminal.go:Shell() - 42ns
[TRACE] 22:05:41.687 path.go:Path() - 42ns
```

--- a/src/template/date.go
+++ b/src/template/date.go
@@ -1,0 +1,58 @@
+package template
+
+import (
+	"strconv"
+	"time"
+)
+
+// dateInZone is a replacement for sprig's dateInZone that adds support for string
+// epoch values. Sprig's unixEpoch returns a string, which sprig's own date functions
+// do not handle — they fall through to time.Now(). This wrapper parses numeric strings
+// as Unix timestamps so that patterns like `{{ now | unixEpoch | date "..." }}` work.
+func dateInZone(fmt string, date interface{}, zone string) string {
+	var t time.Time
+
+	switch v := date.(type) {
+	case time.Time:
+		t = v
+	case *time.Time:
+		t = *v
+	case int64:
+		t = time.Unix(v, 0)
+	case int:
+		t = time.Unix(int64(v), 0)
+	case int32:
+		t = time.Unix(int64(v), 0)
+	case string:
+		if epoch, err := strconv.ParseInt(v, 10, 64); err == nil {
+			t = time.Unix(epoch, 0)
+		} else {
+			t = time.Now()
+		}
+	default:
+		t = time.Now()
+	}
+
+	loc, err := time.LoadLocation(zone)
+	if err != nil {
+		loc, _ = time.LoadLocation("UTC")
+	}
+
+	return t.In(loc).Format(fmt)
+}
+
+func ompDate(fmt string, date interface{}) string {
+	return dateInZone(fmt, date, "Local")
+}
+
+func ompDateInZone(fmt string, date interface{}, zone string) string {
+	return dateInZone(fmt, date, zone)
+}
+
+func ompHTMLDate(date interface{}) string {
+	return dateInZone("2006-01-02", date, "Local")
+}
+
+func ompHTMLDateInZone(date interface{}, zone string) string {
+	return dateInZone("2006-01-02", date, zone)
+}

--- a/src/template/date.go
+++ b/src/template/date.go
@@ -9,7 +9,7 @@ import (
 // epoch values. Sprig's unixEpoch returns a string, which sprig's own date functions
 // do not handle — they fall through to time.Now(). This wrapper parses numeric strings
 // as Unix timestamps so that patterns like `{{ now | unixEpoch | date "..." }}` work.
-func dateInZone(fmt string, date interface{}, zone string) string {
+func dateInZone(fmt string, date any, zone string) string {
 	var t time.Time
 
 	switch v := date.(type) {
@@ -41,18 +41,18 @@ func dateInZone(fmt string, date interface{}, zone string) string {
 	return t.In(loc).Format(fmt)
 }
 
-func ompDate(fmt string, date interface{}) string {
+func ompDate(fmt string, date any) string {
 	return dateInZone(fmt, date, "Local")
 }
 
-func ompDateInZone(fmt string, date interface{}, zone string) string {
+func ompDateInZone(fmt string, date any, zone string) string {
 	return dateInZone(fmt, date, zone)
 }
 
-func ompHTMLDate(date interface{}) string {
+func ompHTMLDate(date any) string {
 	return dateInZone("2006-01-02", date, "Local")
 }
 
-func ompHTMLDateInZone(date interface{}, zone string) string {
+func ompHTMLDateInZone(date any, zone string) string {
 	return dateInZone("2006-01-02", date, zone)
 }

--- a/src/template/date_test.go
+++ b/src/template/date_test.go
@@ -21,10 +21,10 @@ func TestDateFromStringEpoch(t *testing.T) {
 
 	// Use date_in_zone with UTC so tests are timezone-independent.
 	cases := []struct {
+		Context  any
 		Case     string
 		Expected string
 		Template string
-		Context  any
 	}{
 		{
 			Case:     "string epoch via date_in_zone",
@@ -89,10 +89,10 @@ func TestDateAndHTMLDateFunctions(t *testing.T) {
 	epochStr := fmt.Sprintf("%d", knownEpoch)
 
 	cases := []struct {
+		Context  any
 		Case     string
 		Expected string
 		Template string
-		Context  any
 	}{
 		{
 			Case:     "date function with string epoch",

--- a/src/template/date_test.go
+++ b/src/template/date_test.go
@@ -79,4 +79,3 @@ func TestDateFromStringEpoch(t *testing.T) {
 		}
 	}
 }
-

--- a/src/template/date_test.go
+++ b/src/template/date_test.go
@@ -79,3 +79,43 @@ func TestDateFromStringEpoch(t *testing.T) {
 		}
 	}
 }
+
+func TestDateAndHTMLDateFunctions(t *testing.T) {
+	// Override time.Local to UTC so `date` and `htmlDate` produce deterministic output.
+	origLocal := time.Local
+	time.Local = time.UTC
+	defer func() { time.Local = origLocal }()
+
+	epochStr := fmt.Sprintf("%d", knownEpoch)
+
+	cases := []struct {
+		Case     string
+		Expected string
+		Template string
+		Context  any
+	}{
+		{
+			Case:     "date function with string epoch",
+			Expected: "13 Jun 19 20:39 +0000",
+			Template: `{{ date "02 Jan 06 15:04 -0700" .Epoch }}`,
+			Context:  struct{ Epoch string }{Epoch: epochStr},
+		},
+		{
+			Case:     "htmlDate function with zero string epoch",
+			Expected: "1970-01-01",
+			Template: `{{ htmlDate .Epoch }}`,
+			Context:  struct{ Epoch string }{Epoch: "0"},
+		},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.Environment)
+		env.On("Shell").Return("foo")
+		Cache = new(cache.Template)
+		Init(env, nil, nil)
+
+		text, err := Render(tc.Template, tc.Context)
+		assert.NoError(t, err, tc.Case)
+		assert.Equal(t, tc.Expected, text, tc.Case)
+	}
+}

--- a/src/template/date_test.go
+++ b/src/template/date_test.go
@@ -1,0 +1,82 @@
+package template
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/cache"
+	"github.com/jandedobbeleer/oh-my-posh/src/runtime/mock"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// knownEpoch is 2019-06-13 20:39:39 UTC, matching sprig's own test fixture.
+const knownEpoch = int64(1560458379)
+
+func TestDateFromStringEpoch(t *testing.T) {
+	// This is the core regression: unixEpoch returns a string, and piping that
+	// string into `date` must produce the correct formatted date, not time.Now().
+	epochStr := fmt.Sprintf("%d", knownEpoch)
+
+	// Use date_in_zone with UTC so tests are timezone-independent.
+	cases := []struct {
+		Case     string
+		Expected string
+		Template string
+		Context  any
+	}{
+		{
+			Case:     "string epoch via date_in_zone",
+			Expected: "13 Jun 19 20:39 +0000",
+			Template: `{{ date_in_zone "02 Jan 06 15:04 -0700" .Epoch "UTC" }}`,
+			Context:  struct{ Epoch string }{Epoch: epochStr},
+		},
+		{
+			Case:     "int64 epoch via date_in_zone",
+			Expected: "13 Jun 19 20:39 +0000",
+			Template: `{{ date_in_zone "02 Jan 06 15:04 -0700" .Epoch "UTC" }}`,
+			Context:  struct{ Epoch int64 }{Epoch: knownEpoch},
+		},
+		{
+			Case:     "time.Time via date_in_zone",
+			Expected: "13 Jun 19 20:39 +0000",
+			Template: `{{ date_in_zone "02 Jan 06 15:04 -0700" .Epoch "UTC" }}`,
+			Context:  struct{ Epoch time.Time }{Epoch: time.Unix(knownEpoch, 0).UTC()},
+		},
+		{
+			Case:     "string epoch direct call dateInZone",
+			Expected: "13 Jun 19 20:39 +0000",
+			Template: `{{ dateInZone "02 Jan 06 15:04 -0700" .Epoch "UTC" }}`,
+			Context:  struct{ Epoch string }{Epoch: epochStr},
+		},
+		{
+			Case:     "htmlDate with zero string epoch",
+			Expected: "1970-01-01",
+			Template: `{{ htmlDateInZone .Epoch "UTC" }}`,
+			Context:  struct{ Epoch string }{Epoch: "0"},
+		},
+		{
+			Case:     "non-numeric string falls back to current time (not panic)",
+			Template: `{{ date_in_zone "02 Jan 06" .Epoch "UTC" }}`,
+			Context:  struct{ Epoch string }{Epoch: "not-a-number"},
+			// Expected is time-dependent; we just verify no error is returned.
+		},
+	}
+
+	for _, tc := range cases {
+		env := new(mock.Environment)
+		env.On("Shell").Return("foo")
+		Cache = new(cache.Template)
+		Init(env, nil, nil)
+
+		text, err := Render(tc.Template, tc.Context)
+		assert.NoError(t, err, tc.Case)
+		if tc.Expected != "" {
+			assert.Equal(t, tc.Expected, text, tc.Case)
+		} else {
+			assert.NotEmpty(t, text, tc.Case)
+		}
+	}
+}
+

--- a/src/template/func_map.go
+++ b/src/template/func_map.go
@@ -27,6 +27,12 @@ func funcMap() template.FuncMap {
 		"stat":         stat,
 		"dir":          filepath.Dir,
 		"base":         filepath.Base,
+		// Override sprig date functions to support string epoch values (e.g. output of unixEpoch).
+		"date":           ompDate,
+		"date_in_zone":   ompDateInZone,
+		"dateInZone":     ompDateInZone,
+		"htmlDate":       ompHTMLDate,
+		"htmlDateInZone": ompHTMLDateInZone,
 	}
 
 	for key, fun := range sprig.TxtFuncMap() {

